Std/Data/List/Basic.lean

```diff
@@ -823,6 +823,36 @@ inductive Forall₂ (R : α → β → Prop) : List α → List β → Prop
 
 attribute [simp] Forall₂.nil
 
+theorem forall₂_cons {R : α → β → Prop} {a b as bs} :
+    Forall₂ R (a :: as) (b :: bs) ↔ R a b ∧ Forall₂ R as bs :=
+  ⟨fun | .cons h tail => ⟨h, tail⟩, fun ⟨head, tail⟩ => .cons head tail⟩
+
+/--
+Check if all elements in the first List are `R` related to the element at the same position in the
+second List.
+-/
+def all₂ (R : α → β → Prop) [∀ a b, Decidable (R a b)] : List α → List β → Bool
+  | [], [] => true
+  | a::as, b::bs =>
+    if R a b then
+      all₂ R as bs
+    else false
+  | _, _ => false
+
+theorem all₂_iff_forall₂ {R : α → β → Prop} [∀ a b, Decidable (R a b)] :
+    ∀ l₁ l₂, all₂ R l₁ l₂ ↔ Forall₂ R l₁ l₂
```

Comment:
```suggestion
@[simp]
theorem all₂_eq_true {R : α → β → Prop} [∀ a b, Decidable (R a b)] :
    ∀ l₁ l₂, all₂ R l₁ l₂ ↔ Forall₂ R l₁ l₂
```
to match `List.all_eq_true`

Reply:
yeah, makes more sense when viewing the generated docs :+1: